keras_nlp/src/models/resnet_vd/resnet_vd_backbone.py

```diff
@@ -0,0 +1,381 @@
+        )(x)
+
+        x = keras.layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=BN_EPSILON, name="conv1_bn"
+        )(x)
+        x = keras.layers.Activation("relu", name="conv1_relu")(x)
+
+        x = keras.layers.Conv2D(
+            32, 3, strides=1, use_bias=False, padding="same", name="conv2_conv"
+        )(x)
+
+        x = keras.layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=BN_EPSILON, name="conv2_bn"
+        )(x)
+        x = keras.layers.Activation("relu", name="conv2_relu")(x)
+
+        x = keras.layers.Conv2D(
+            64, 3, strides=1, use_bias=False, padding="same", name="conv3_conv"
+        )(x)
+
+        x = keras.layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=BN_EPSILON, name="conv3_bn"
+        )(x)
+        x = keras.layers.Activation("relu", name="conv3_relu")(x)
+
+        x = keras.layers.MaxPooling2D(
+            3, strides=2, padding="same", name="pool1_pool"
+        )(x)
+
+        num_stacks = len(stackwise_filters)
+
+        pyramid_level_inputs = {}
+        for stack_index in range(num_stacks):
+            x = apply_stack(
+                x,
+                filters=stackwise_filters[stack_index],
+                blocks=stackwise_blocks[stack_index],
+                stride=stackwise_strides[stack_index],
+                block_type=block_type,
+                first_shortcut=(block_type == "block" or stack_index > 0),
+                name=f"vd_stack_{stack_index}",
+            )
+            pyramid_level_inputs[f"P{stack_index + 2}"] = (
+                x._keras_history.operation.name
+            )
+
+        # Create model.
+        super().__init__(inputs=inputs, outputs=x, **kwargs)
+
+        # All references to `self` below this line
+        self.pyramid_level_inputs = pyramid_level_inputs
+        self.stackwise_filters = stackwise_filters
+        self.stackwise_blocks = stackwise_blocks
+        self.stackwise_strides = stackwise_strides
+        self.include_rescaling = include_rescaling
+        self.block_type = block_type
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "stackwise_filters": self.stackwise_filters,
+                "stackwise_blocks": self.stackwise_blocks,
+                "stackwise_strides": self.stackwise_strides,
+                "include_rescaling": self.include_rescaling,
+                # Remove batch dimension from `input_shape`
+                "input_shape": self.input_shape[1:],
+                "block_type": self.block_type,
+            }
+        )
+        return config
+
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return copy.deepcopy(backbone_presets)
+
+    @classproperty
+    def presets_with_weights(cls):
+        """Dictionary of preset names and configurations that include
+        weights."""
+        return copy.deepcopy(backbone_presets_with_weights)
```

Comment:
we shouldn't need to do this anymore

Reply:
@mattdangerw PTAL https://github.com/keras-team/keras-nlp/pull/1766#discussion_r1717086972